tests/stability/soak_parallel_rm.sh

```diff
@@ -0,0 +1,208 @@
+#!/bin/bash
+#
+# Copyright (c) 2017-2018, 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test will run a number of parallel containers, and then try to
+# 'rm -f' them all at the same time. It will check after each run and
+# rm that we have the expected number of containers, shims,
+# qemus and runtimes active
+# The goals are two fold:
+# - spot any stuck or non-started components
+# - catch any hang ups
+
+cidir=$(dirname "$0")
+source "${cidir}/../metrics/lib/common.bash"
+source "/etc/os-release" || source "/usr/lib/os-release"
+set -x
+
+# How many times will we run the test loop...
+ITERATIONS="${ITERATIONS:-5}"
+
+# the system 'free available' level where we stop running the tests, as otherwise
+#  the system can crawl to a halt, and/or start refusing to launch new VMs anyway
+# We choose 2G, as that is one of the default VM sizes for Kata
+MEM_CUTOFF="${MEM_CUTOFF:-(2*1024*1024*1024)}"
+
+# do we need a command argument for this payload?
+COMMAND="${COMMAND:-tail -f /dev/null}"
+
+# Runtime path
+RUNTIME_PATH=$(command -v $RUNTIME)
+
+# The place where virtcontainers keeps its active pod info
+# This is ultimately what 'kata-runtime list' uses to get its info, but
+# we can also check it for sanity directly
+VC_POD_DIR="${VC_POD_DIR:-/run/vc/sbs}"
+
+# let's cap the test. If you want to run until you hit the memory limit
+# then just set this to a very large number
+MAX_CONTAINERS="${MAX_CONTAINERS:-110}"
+
+KATA_HYPERVISOR="${KATA_HYPERVISOR:-qemu}"
+
+check_vsock_active() {
```

Comment:
to follow the uniformity used in the `tests` set, you could use the keyword `function` to define a function :)